test/parallel/test_torch.py

```diff
@@ -229,6 +229,87 @@ def test_horovod_allreduce_average(self):
 
             assert torch.allclose(averaged, tensor, threshold), 'hvd.allreduce produces incorrect results'
 
+    def test_horovod_allreduce_min(self):
+        """Test that the allreduce correctly minimizes 1D, 2D, 3D tensors."""
+        hvd.init()
+        size = hvd.size()
+        rank = hvd.rank()
+        dtypes = self.filter_supported_types([torch.IntTensor, torch.LongTensor,
+                     torch.FloatTensor, torch.DoubleTensor])
```

Comment:
Is there a reason not to include `torch.HalfTensor` (FP16 on CPU) in the three new tests for PyTorch?

Reply:
Not that I know of, but the new tests I added just copied the existing allreduce tests (including the data type coverage). We can extend this in a different PR if you think it is worthwhile.

Reply:
Ah, on a closer look some tests include the `torch.HalfTensor`, others don't, so it's a bit random really, which tests have been updated and which haven't. 

Most TF tests cover fp16, so it's not that urgent really.